### DISCUSSION TITLE
Arcana Logins support

### DISCUSSION THREAD

**Discussion by MouneeswarMedepally** [arcana-network/discussions#25] [Arcana Dashboard]
I have added client ids and keys in arcana dashboard with redirect URI proivded by arcana. But at the time of login to my application, I am getting 500 internal server error on redirect call. Can someone help me to resolve this issue?

Also when i click on transak in arcana wallet to purchase crypto , it is redirecting to transak with undefined APIkey

**Comment by mmjee** [comment]
On Monday, June 10 2024 Anno Domini 10:24 AM India Standard Time, MouneeswarMedepally wrote:
> I have added client ids and keys in arcana dashboard with redirect URI proivded by arcana. But at the time of login to my application, I am getting 500 internal server error on redirect call. Can someone help me to resolve this issue?
> 
> Also when i click on transak in arcana wallet to purchase crypto , it is redirecting to transak with undefined APIkey 
> 
What OAuth provider (Google, Twitter, etc.) and environment (test-net, main-net, etc.) were you trying to use? The Transak problem may be fixed now.
-- 
Regards,

**Reply by MouneeswarMedepally** [reply]
I am using google OAuth2.0, twitter's default project's client id and key, discord's OAuth2 with test-net environment

**Reply by MouneeswarMedepally** [reply]
Also, transak is still showing undefined API key, is it due to arcana test-net ?

**Reply by MouneeswarMedepally** [reply]
Can someone help me to solve above issues?

**Reply by MouneeswarMedepally** [reply]
Google and discord logins are working fine now in testnet , but getting internal server error for twitter login and facing same undefined API key issue for transak

**Comment by mmjee** [comment]
On Monday, June 10 2024 Anno Domini 03:26 PM India Standard Time, MouneeswarMedepally wrote:
> Google and discord logins are working fine now in testnet , but getting internal server error for twitter login and facing same undefined API key issue for transak
> 
> 
I’m unable to reproduce the Transak undefined API key problem, please clear your cache and check again. ‘Internal server errors’ in the Twitter log-in process usually indicates that the credentials are not being accepted by Twitter.
-- 
Regards,

**Reply by MouneeswarMedepally** [reply]
when i click on login with twitter, it is showing that error instead of opening window for login
And for transak : 
it is redirecting to following url: 
https://global.transak.com/**?apiKey=undefined**&walletAddress=
and with below error:
Bad Request: Invalid API key. Please double check your API key and your "environment" name. If you are entering the staging API Key, then your "environment" should be staging

**Reply by MouneeswarMedepally** [reply]
Issues resolved , thank you for your support!
